File: frontend/src/api/sprint.ts
import api from './axios'

export interface Sprint {
  id: number
  projectId: number
  name: string
  goal: string
  startDate: string
  endDate: string
  status: string
  totalCapacity: number
  createdAt: string
}

export interface CreateSprintRequest {
  projectId: number
  name: string
  goal?: string
  startDate: string
  endDate: string
  totalCapacity?: number
}

export const sprintApi = {
  getByProject: (projectId: number) => api.get<Sprint[]>(`/sprints/project/${projectId}`),
  getById: (id: number) => api.get<Sprint>(`/sprints/${id}`),
  create: (data: CreateSprintRequest) => api.post<Sprint>('/sprints', data),
  start: (id: number) => api.post<Sprint>(`/sprints/${id}/start`),
  complete: (id: number) => api.post<Sprint>(`/sprints/${id}/complete`),
}

export const getSprints = async (projectId: number) => {
  const response = await sprintApi.getByProject(projectId)
  return response.data
}

export const createSprint = async (projectId: number, data: Omit<CreateSprintRequest, 'projectId'>) => {
  const response = await sprintApi.create({ ...data, projectId })
  return response.data
}

File: burndown-backend-ms/project-service/src/main/java/com/burndown/project/entity/Sprint.java
package com.burndown.project.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "sprints", schema = "pg_project")
@EntityListeners(AuditingEntityListener.class)
public class Sprint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "project_id", nullable = false)
    private Long projectId;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String goal;

    @Column(name = "start_date")
    private LocalDate startDate;

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(length = 20)
    private String status = "PLANNED";

    @Column(name = "total_capacity", precision = 10, scale = 2)
    private BigDecimal totalCapacity;

    @Column(name = "committed_points", precision = 10, scale = 2)
    private BigDecimal committedPoints;

    @Column(name = "completed_points", precision = 10, scale = 2)
    private BigDecimal completedPoints = BigDecimal.ZERO;

    @Column(precision = 10, scale = 2)
    private BigDecimal velocity;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "started_at")
    private LocalDateTime startedAt;

    @Column(name = "completed_at")
    private LocalDateTime completedAt;
}
